<launch>

  <node pkg="loam_velodyne" type="scanRegistration" name="scanRegistration" output="screen"/>
  <node pkg="loam_velodyne" type="laserOdometry" name="laserOdometry" output="screen"/>
  <node pkg="loam_velodyne" type="laserMapping" name="laserMapping" output="screen"/>
  <node pkg="loam_velodyne" type="transformMaintenance" name="transformMaintenance" output="screen">
    <param name="my_pose_enable" type="bool" value="false" />
  </node>

  
  <!-- ed: poseInitializerVelo 사용 -->
  <node pkg="loam_velodyne" type="poseInitializerVelo" name="poseInitializer" output="screen">
    <param name="enable_init_map" type="bool" value="true" />
    <param name="map_file" type="string" value="original.pcd" /> <!-- change here ! /-->
    <param name="init_map_record_time" type="double" value="1.0" />
    <param name="file_debug" type="bool" value="true" />
  </node>

  <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find loam_velodyne)/loam_velodyne_with_init.rviz"/>

</launch>
